cluster/cluster.yaml

```diff
@@ -1340,8 +1340,61 @@ Resources:
           PolicyName: root
       RoleName: "{{.Cluster.LocalID}}-kube-metrics-adapter"
     Type: 'AWS::IAM::Role'
-{{ if eq .Cluster.Environment "e2e" }}
-  # This is a hack to easily create an aws iam role and s3 bucket for testing
+
+  EBSCSIControllerIAMRole:
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action:
+              - 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+          - Action:
+              - 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              AWS: !Join
+                - ''
+                - - 'arn:aws:iam::{{.Cluster.InfrastructureAccount | getAWSAccountID}}:role/'
+                  - !Ref MasterIAMRole
+          - Action:
+              - 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              AWS: !Join
+                - ''
+                - - 'arn:aws:iam::{{.Cluster.InfrastructureAccount | getAWSAccountID}}:role/'
+                  - !Ref WorkerIAMRole
```

Comment:
Do we really need it? We'll run the controller on the masters, so the workers shouldn't be able to delete volumes.

Reply:
So the worker role is necessary right now because of how `kube-aws-iam-controller` gets credentials right now. It assumes the worker's role first to keep compatibility with kube2iam. So the `WorkerIAMRole` needs to be present for now. Once we move to pod identity credentials then this can be removed.

Reply:
I see, thanks!

Reply:
Can you add a note that this can be removed as a comment somewhere (Also fine in the pod identity issue)? Otherwise I think it'll just stick around b/c no one knows what is used for what.